pkg/cmd/render/render.go

```diff
@@ -80,28 +136,98 @@ func (r *renderOpts) Complete() error {
 type TemplateData struct {
 	genericrenderoptions.ManifestConfig
 	genericrenderoptions.FileConfig
+
+	clusterInfrastructure  configv1.Infrastructure
+	cloudProviderConfigMap corev1.ConfigMap
 }
 
 // Run contains the logic of the render command.
 func (r *renderOpts) Run() error {
 	renderConfig := TemplateData{}
+
+	// XXX: What is this file used for?
 	if len(r.clusterConfigFile) > 0 {
 		_, err := ioutil.ReadFile(r.clusterConfigFile)
 		if err != nil {
 			return err
 		}
 		// TODO I'm thinking we parse this into a map and reference it that way
 	}
+
+	if len(r.clusterInfrastructureFile) > 0 && len(r.cloudProviderConfigInputFile) > 0 && len(r.cloudProviderConfigOutputFile) > 0 {
+
+		// Read, parse, and save the infrastructure object
+		var clusterInfrastructure configv1.Infrastructure
+		clusterInfrastructureFileData, err := ioutil.ReadFile(r.clusterInfrastructureFile)
```

Comment:
use short variable names. This looks like Java code. `bs` for bytes is commonly used.

Reply:
I've shortened the names. I am by no means a Java programmer, was just following conventions used in much of the golang code I've reviewed ;-)